include/cache.h

```diff
@@ -13,59 +13,185 @@
 extern "C" {
 #endif
 
-void arch_dcache_flush(void *addr, size_t size);
-void arch_dcache_invd(void *addr, size_t size);
+/**
+ * Common operations for the caches
+ */
+#define K_CACHE_WB	BIT(0)
+#define K_CACHE_INVD	BIT(1)
+#define K_CACHE_WB_INVD	(K_CACHE_WB | K_CACHE_INVD)
```

Comment:
Could add a few word describing the intention of the operations here, or make reference to a description. Previous discussions in this PR has identified that there might be different interpretations and terminology depending on context.
Something along the lines of
* _"WB means write-back and intends to transfer dirty cache lines to memory in a copy-back cache policy. May be a no-op in write-back cache policy."_
* _"INVD means invalidate and will mark cache lines as not valid. A future access to the associated address is guaranteed to generate a memory fetch."_

Reply:
ACK